From d6f5f7fc273f41b22fde9aea2d5385a0fb829e3e Mon Sep 17 00:00:00 2001
From: Elijah Shakkour <elijahsh@mellanox.com>
Date: Sun, 14 Jul 2019 14:16:05 +0300
Subject: [PATCH] mlx5: Set multifunction bit in config space header-type for
 PF

Issue: 1824873

According to PCI spec, this bit should be set to indicate
that device supports multifunction.

Change-Id: Ic5174af0f27f58fe31f172a806e461b635391316
Signed-off-by: Elijah Shakkour <elijahsh@mellanox.com>
Signed-off-by: Leon Romanovsky <leonro@mellanox.com>
---
 hw/net/mellanox/mlx5/mlx5.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/hw/net/mellanox/mlx5/mlx5.c b/hw/net/mellanox/mlx5/mlx5.c
index 6699ae23..34c56cea 100644
--- a/hw/net/mellanox/mlx5/mlx5.c
+++ b/hw/net/mellanox/mlx5/mlx5.c
@@ -289,6 +289,10 @@ static int mlx5_pci_config_space_init(PCIDevice *pci_dev)
     pci_set_word(pci_dev->wmask + PCI_STATUS, MLX5_CONFIG_SPACE_STATUS_MASK);
 
     if (s->function_id == MLX5_PF_FUNCTION_ID) {
+        /* set multifunction bit in PCI_HEADER_TYPE */
+        /* bits 0-6: header type
+         * bit 7: multi/single function device */
+        pci_dev->config[PCI_HEADER_TYPE] |= 0x80;
         /* change access permissions of R/W PCI COMMAND fields to R/W */
         pci_set_word(pci_dev->wmask + PCI_COMMAND, MLX5_CONFIG_SPACE_COMMAND_MASK);
 
-- 
2.20.1

